cg/meta/archive/ddn_dataflow.py

```diff
@@ -86,15 +77,24 @@ def dict(self, **kwargs) -> dict:
         payload["metadataList"] = []
         return payload
 
-    def post_request(self, url: str, headers: dict) -> bool:
+    def post_request(self, url: str, headers: dict) -> int:
         """Sends a request to the given url with, the given headers, and its own content as
-        payload."""
-        return APIRequest.api_request_from_content(
+        payload. Raises an error if the response code is not ok. Returns the job ID of the
+        launched transfer task.
+        """
+        response: Response = APIRequest.api_request_from_content(
             api_method=APIMethods.POST,
             url=url,
             headers=headers,
             json=self.dict(),
-        ).ok
+        )
+        response.raise_for_status()
+        parsed_response = TransferResponse(
```

Comment:
I think Pydantic has some nice functionality for parsing the response body into a data model. Just an alternative to consider which might be easier to read, something like
```python
TransferResponse.mode_validate(response.json())
```

Reply:
Also, maybe an alternative name for the model could just be `TransferJob` instead of `TransferResponse`? I think it might be confusing to include `Response` in the model name if we settle on using a pattern where we return a data model from the clients

Reply:
Implemented. It was only added in pydantic v2 though so I hade to make some other changes as well